### DISCUSSION TITLE
How to skip import in dry mode?

### DISCUSSION THREAD

**Discussion by djasa** [behave/behave#1093] [Q&A] [answered]
I would like to be able to run `behave --dry -f steps.usage` on machines that aren't expected to run the actual scenarios and thus don't need all the runtime dependencies. What's the cleanest way to find out behave's `dry_run` value from within `features/environment.py` or a module that the environment.py loads, so I could do something like this?

```python
from behave import some_behave_submodule

if not some_behave_submodule.dry_run:
    import wet_only_dependency
```

**Comment by jenisys** [comment, marked as answer]
Such a functionality is currently not supported (on module level) while loading the environment-file and the step-modules.
In addition, I do not think that this is not such a good idea. It is a HACK in my opinion (using the dry-run mode and having a partially usable workspace are two different things).

If  you need such a functionality, you can easily support your use case by using a special environment variable, like:

```bash
# -- EXAMPLE bash-like shell is used
$ BEHAVE_SLIM_MODE=ON behave --dry -f steps.usage
...
```

```python
# -- FILE: features/environment.py
import os
BEHAVE_SLIM_MODE_ENABLED = os.environ.get("BEHAVE_SLIM_MODE", "OFF") == "ON"
if not BEHAVE_SLIM_MODE_ENABLED:
    # -- NORMAL CASE: Import all needed modules and parts.
    ...
else:
    ...  # Whatever 
```
